featurebyte/session/session_helper.py

```diff
@@ -21,6 +26,52 @@ def _to_query_str(query: Union[str, Expression], source_type: SourceType) -> str
     return sql_to_string(query, source_type)
 
 
+async def validate_output_row_index(session: BaseSession, output_table_name: str) -> None:
+    """
+    Validate row index is unique in the generated output table
+
+    Parameters
+    ----------
+    session: BaseSession
+        Database session
+    output_table_name: str
+        Name of the table to be validated
+
+    Raises
+    ------
+    ValueError
+        If any of the row index value is not unique
+    """
+    query = sql_to_string(
+        expressions.select(
+            expressions.alias_(
+                expressions.Max(this=quoted_identifier("row_index_count")),
+                alias="max_row_index_count",
+                quoted=True,
+            )
+        ).from_(
+            expressions.select(
+                expressions.alias_(
+                    expressions.Count(this=expressions.Star()),
+                    alias="row_index_count",
+                    quoted=True,
+                ),
+            )
+            .from_(quoted_identifier(output_table_name))
+            .group_by(InternalName.TABLE_ROW_INDEX)
```

Comment:
Would using a count distinct on row index vs count(*) check be more efficient?

Reply:
Possibly. The query will be simpler to read. Let me test both versions.